### DISCUSSION TITLE
Multiple dynamic routes

### DISCUSSION THREAD

**Discussion by oogxdd** [vercel/next.js#17660] [Help] [answered]
# Feature request

## Is your feature request related to a problem? Please describe.

So for nice-looking urls I want to make them as short as possible. eg i want `/@ignatif` url for users and `/how-to-...` for articles. Due to Next.js routing nature I can't achieve it as I can't create both `[user].js` and `[article].js` files in a same folder

## Describe the solution you'd like

When hitting `/@ignatif` url I want to land on a user page
When hitting `/how-to-...` url I want to land on an article page

I want to maybe be able to create multiple files like
`/@[user].js`
`/[article].js`

in a same folder so each time i land on url starting with "@" i end up on a user page, otherwise on an article page

## Describe alternatives you've considered

I achieved my goal by having following structure

`/[slug].js`
`/articles/[article].js`
`/users/[user].js`

in a `/[slug].js` I have:

```javascript
All.getInitialProps = ({ res, query }) => {
  if (query.slug.startsWith("@")) {
    res.writeHead(302, { Location: `/users/${query.slug}` });
  } else {
    res.writeHead(302, { Location: `/articles/${query.slug}` });
  }
  res.end();
};
```

and in `/articles/[article].js` and `/users/[user].js` I have:

```javascript
  useEffect(() => {
    if (process.browser) {
      const username = window.location.pathname.substring(7); // remove `/users/`
      window.history.replaceState({}, null, `/${username}`);
    }
  }, [router]);
```

however it causes a quick blink in a browser url bar which spoils the whole experience

Example can be found here: https://nextjs-multiple-dynamic-urls.vercel.app/

## Additional context

Currently that's the only thing that's stopping me from making a transition to Next.js, which works perfectly for my use case as my application heavily relies on social sharing an SEO. But convenient urls is my priority #1

It even stops me from using Vercel (which i ❤️) as I'm going to implement Open Graph tags using solution like https://prerender.io/ which requires serving the site from a custom Node server

Thanks

**Comment by lfades** [comment, marked as answer]
[Next.js Rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) should be able to cover your use case by using [Regex Patch Matching](https://nextjs.org/docs/api-reference/next.config.js/rewrites#regex-path-matching) 👍
